### PR TITLE
APIS-3739 - Privacy and whitelisting

### PR DESCRIPTION
**Pull request by rose-garratt**

### CODE REVIEW

File: app/uk/gov/hmrc/apidocumentation/controllers/DocumentationController.scala [hunk 1]
```diff
@@ -36,6 +36,9 @@ import uk.gov.hmrc.play.frontend.controller.FrontendController
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
+
```
Comment (mitchellb1) [inline app/uk/gov/hmrc/apidocumentation/controllers/DocumentationController.scala]: Too much whitespace?

File: app/uk/gov/hmrc/apidocumentation/models/Documentation.scala [hunk 2]
```diff
@@ -26,7 +26,7 @@ object APIAccessType extends Enumeration {
   val PRIVATE, PUBLIC = Value
 }
 
-case class APIAccess(`type`: APIAccessType.Value)
+case class APIAccess(`type`: APIAccessType.Value, isTrial: Option[Boolean] = None)
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apidocumentation/models/Documentation.scala]: This `isTrial` flag can have three states - is that intended?

File: app/uk/gov/hmrc/apidocumentation/models/Documentation.scala [hunk 3]
```diff
@@ -238,4 +243,10 @@ case class ServiceDetails(serviceName: String, serviceUrl: String)
 
 case class RawDocumentationContent(contentType: String, content: String)
 
-case class ErrorResponse(code: Option[String] = None, message: Option[String] = None)
\ No newline at end of file
+case class ErrorResponse(code: Option[String] = None, message: Option[String] = None)
+
+object DocsVisibility extends Enumeration {
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apidocumentation/models/Documentation.scala]: Scala native Enumerations has issues - consider alternative like Enumeratum. See TPDF work that I did to change Enumerations.

https://github.com/hmrc/third-party-developer-frontend/commit/b4a1cfbfd28b35b19d0befa2a9f69aff60ce8157

File: app/uk/gov/hmrc/apidocumentation/controllers/DocumentationController.scala (hunk 1 above)
Comment (rose-garratt) [reply app/uk/gov/hmrc/apidocumentation/controllers/DocumentationController.scala]: Agreed

File: app/uk/gov/hmrc/apidocumentation/models/Documentation.scala (hunk 2 above)
Comment (rose-garratt) [reply app/uk/gov/hmrc/apidocumentation/models/Documentation.scala]: That's known and tested for. IsTrial may not be present and doesn't make sense for PUBLIC apis.

File: app/uk/gov/hmrc/apidocumentation/models/Documentation.scala (hunk 3 above)
Comment (rose-garratt) [reply app/uk/gov/hmrc/apidocumentation/models/Documentation.scala]: Love to, should do as part of a larger piece of work though?

**Pull request merged**